mesonbuild/linkers.py

```diff
@@ -1372,3 +1372,88 @@ def get_soname_args(self, env: 'Environment', prefix: str, shlib_name: str,
                         suffix: str, soversion: str, darwin_versions: T.Tuple[str, str],
                         is_shared_module: bool) -> T.List[str]:
         return []
+
+class TCCDynamicLinker(PosixDynamicLinkerMixin, DynamicLinker):
+    """Linker for TCC compiler."""
+    id = 'tcc'
+
+    def __init__(self, for_machine: mesonlib.MachineChoice,
+                 *, version: str = 'unknown version'):
+        super().__init__(['tcc'], for_machine, '-Wl,', [],
+                         version=version)
+
+    def get_allow_undefined_args(self) -> T.List[str]:
+        return []
+
+    def get_pic_args(self) -> T.List[str]:
+        return []
+
+    def get_link_whole_for(self, args: T.List[str]) -> T.List[str]:
+        if not args:
+            return args
+        return self._apply_prefix('--whole-archive') + args + self._apply_prefix('--no-whole-archive')
+
+    def get_coverage_args(self) -> T.List[str]:
+        return []
+
+    def export_dynamic_args(self, env: 'Environment') -> T.List[str]:
+        m = env.machines[self.for_machine]
+        if m.is_windows() or m.is_cygwin():
+            return self._apply_prefix('--export-all-symbols')
+        return self._apply_prefix('-export-dynamic')
+
+    def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
```

Comment:
This section should likely be re-considered; I'm not sure of SunOS or cygwin parts, quite likely the logic can be simplified.

Reply:
I removed the SunOS part, but honestly i don't know how to improve this further without breaking stuff. I copied most of the body from gnu IIRC.